x/compute/client/rest/tx.go

```diff
@@ -166,3 +182,30 @@ func executeContractHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
 		utils.WriteGenerateStdTxResponse(w, cliCtx, req.BaseReq, []sdk.Msg{msg})
 	}
 }
+
+func GetCodeHashByContractAddr(cliCtx context.CLIContext, contractAddr sdk.AccAddress) ([]byte, error) {
+	route := fmt.Sprintf("custom/%s/%s/%s", types.QuerierRoute, keeper.QueryContractHash, contractAddr.String())
+	res, _, err := cliCtx.Query(route)
+	if err != nil {
+		return nil, err
+	}
+
+	return []byte(hex.EncodeToString(res)), nil
+}
```

Comment:
feels like this pattern keeps repeating. is it not close enough to wrap up in a function to prevent the duplication?

Reply:
How would you do it differently?

Reply:
I'd create a wrapper for `context.CLIContext` (I understand that it comes from another package?) and then implement these queries as methods on it. that way you have an obvious place to write all the different queries you need to do, and you get one thing that exposes all available queries in a structured way.

Reply:
Ah, that could work. You'd have to create the new type from CLIContext every time you wanted to use the functions though, but if there are a few of them it might be worth it. In this case I deleted this since it was unnecessary anyway, but we can keep this is mind if we ever need to add more

Reply:
:+1: